src/services/ckanapi.ts

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+import axios from 'axios';
+import { Dataset } from './../interfaces/dataset.interface';
+import { PackageSearchOptions, PackageSearchResult } from './../interfaces/packageSearch.interface';
+
+interface RawDataset {
+  [key: string]: string | string[] | null;
+}
+
+export default class CKAN {
```

Comment:
This class is a service, right?

It would be more readable if you have a class per use case and name the file and class according to it, like you did with the test.

Reply:
Indeed, this could be a way, but I particularly prefer composition over inheritance, but I don't believe there will be enough common code to justify any of the two solutions.

Reply:
ok for now lets, have just one class PackageSearch and we will see in future if we need composition/inheritence to extract any common code